pkg/util/kvgraph/util.go

```diff
@@ -116,27 +116,37 @@ func addLauncherPod(vmiName, vmiNamespace string, resources []velero.ResourceIde
 	return addVeleroResource(pod.GetName(), vmiNamespace, "pods", resources), nil
 }
 
-func addInstanceType(instanceType v1.InstancetypeMatcher, namespace string, resources []velero.ResourceIdentifier) []velero.ResourceIdentifier {
-	instanceKind := strings.ToLower(instanceType.Kind)
-	switch instanceKind {
-	case "virtualmachineclusterinstancetype":
-		resources = addVeleroResource(instanceType.Name, "", instanceKind, resources)
-	case "virtualmachineinstancetype":
-		resources = addVeleroResource(instanceType.Name, namespace, instanceKind, resources)
+func addInstanceType(vm *v1.VirtualMachine, resources []velero.ResourceIdentifier) []velero.ResourceIdentifier {
+	if vm.Spec.Instancetype != nil {
+		return addInstanceTypeMatcherResource(vm.Spec.Instancetype, vm.Status.InstancetypeRef, vm.GetNamespace(), resources)
 	}
-	resources = addVeleroResource(instanceType.RevisionName, namespace, "controllerrevisions", resources)
 	return resources
 }
 
-func addPreferenceType(preference v1.PreferenceMatcher, namespace string, resources []velero.ResourceIdentifier) []velero.ResourceIdentifier {
-	preferenceKind := strings.ToLower(preference.Kind)
-	switch preferenceKind {
-	case "virtualmachineclusterpreference":
-		resources = addVeleroResource(preference.Name, "", preferenceKind, resources)
-	case "virtualmachinepreference":
-		resources = addVeleroResource(preference.Name, namespace, preferenceKind, resources)
+func addPreference(vm *v1.VirtualMachine, resources []velero.ResourceIdentifier) []velero.ResourceIdentifier {
+	if vm.Spec.Preference != nil {
```

Comment:
since you are checking 	
if matcher == nil {
￼		return resources
￼	}
in line 134, you can just do return here of 
return addInstanceTypeMatcherResource(vm.Spec.Preference, vm.Status.PreferenceRef, vm.GetNamespace(), resources)
same for instanceType